patches/@typechain+ethers-v5+9.0.0.patch

```diff
@@ -0,0 +1,69 @@
+diff --git a/node_modules/@typechain/ethers-v5/dist/codegen/index.js b/node_modules/@typechain/ethers-v5/dist/codegen/index.js
```

Comment:
this patch does 2 things:
1. removes bytecode from all factories factories (unused by us and bloats the bundle-size massively)
2. types `contractName` as `string | <actualName>` so it's usable for extending / typing stuff in TS

Reply:
Instead of patching (1) I think we should pass in `abi` without bytecode embedded in the file, so it doesn't get picked up by typechain.
we could use hardhat abi plugin to export abi, then run typechain through the abis. currently it's going through artifacts.

Reply:
(2) https://github.com/dethcrypto/TypeChain/issues/601 ?